src/app/page.tsx

```diff
@@ -18,11 +37,60 @@ export default function HomePage() {
           secondaries.
         </p>
         <div className="flex items-center gap-2">
-          <Button size="lg" variant="default">
-            <Link href={constants.twitter.url} className="hover:no-underline">
-              Join the waitlist
-            </Link>
-          </Button>
+          <Dialog open={modalOpen}>
+            <DialogTrigger asChild>
+              <Button
+                onClick={() => setModalOpen(true)}
+                className="hover:no-underline"
+              >
+                Join the waitlist
+              </Button>
+            </DialogTrigger>
+            <DialogContent className="sm:max-w-[425px]">
+              <form
+                onSubmit={async (event) => {
```

Comment:
Can we go ahead with a library from the start? Like react-hook-form? The earlier the better

Reply:
good suggestion, let me do that